#! /bin/sh -e

if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch $pdir -f --no-backup-if-mismatch -p1 < $0;;
    -unpatch) patch $pdir -f --no-backup-if-mismatch -R -p1 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

# DP: rename gud.el to gud-bashdb.el

--- bashdb/debugger/emacs/gud.el~	2003-11-10 23:45:38.000000000 +0100
+++ bashdb/debugger/emacs/gud.el	2003-11-10 23:48:18.000000000 +0100
@@ -2644,6 +2644,6 @@
 	  (t nil)))
      (t nil))))
 
-(provide 'gud)
+(provide 'gud-bashdb)
 
 ;;; gud.el ends here
--- bashdb/debugger/emacs/bashdb.el~	2003-11-10 23:45:38.000000000 +0100
+++ bashdb/debugger/emacs/bashdb.el	2003-11-10 23:47:48.000000000 +0100
@@ -28,7 +28,7 @@
 ;; 2. Do M-x bashdb
 
 ;; Codes:
-(require 'gud)
+(require 'gud-bashdb)
 ;; ======================================================================
 ;; bashdb functions
 
